executor/index_merge_reader.go

```diff
@@ -250,7 +255,7 @@ func (e *IndexMergeReaderExecutor) startPartialIndexWorker(ctx context.Context,
 
 func (e *IndexMergeReaderExecutor) buildPartialTableReader(ctx context.Context, workID int) Executor {
 	tableReaderExec := &TableReaderExecutor{
-		baseExecutor: newBaseExecutor(e.ctx, e.schema, 0),
+		baseExecutor: newBaseExecutor(e.ctx, e.schema, e.getIndexID(workID)),
```

Comment:
should use `e.getTableID` ?

Reply:
I found that if use `e.getTableID`, the execution info for **TableRangeScan** will show the empty result. Just like this(`time:0ns, loops:0`):
```
mysql> explain analyze select /*+ use_index_merge(t1, primary, t1a) */ * from t1 where id < 2 or a > 4;
+-------------------------------+---------+---------+-----------+------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------------------------------------+-----------+------+
| id                            | estRows | actRows | task      | access object          | execution info                                                                                                                                                                | operator info                                  | memory    | disk |
+-------------------------------+---------+---------+-----------+------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------------------------------------+-----------+------+
| IndexMerge_8                  | 2.78    | 2       | root      |                        | time:1.147108ms, loops:2, index_task:148.651µs, table_task:{num:2, concurrency:5, time:3.317293ms}                                                                            |                                                | N/A       | N/A  |
| ├─TableRangeScan_5(Build)     | 1.67    | 0       | cop[tikv] | table:t1               | time:0ns, loops:0                                                                                                                                                             | range:[-inf,2), keep order:false, stats:pseudo | N/A       | N/A  |
| ├─IndexRangeScan_6(Build)     | 1.67    | 0       | cop[tikv] | table:t1, index:t1a(a) | time:153.885µs, loops:1, cop_task: {num: 1, max:165.972µs, proc_keys: 0, rpc_num: 1, rpc_time: 101.44µs, copr_cache_hit_ratio: 0.00}                                          | range:(4,+inf], keep order:false, stats:pseudo | N/A       | N/A  |
| └─TableRowIDScan_7(Probe)     | 2.78    | 2       | cop[tikv] | table:t1               | time:877.307µs, loops:5, cop_task: {num: 3, max: 232.059µs, min: 139.22µs, avg: 189.863µs, p95: 232.059µs, rpc_num: 3, rpc_time: 397.616µs, copr_cache_hit_ratio: 0.00}       | keep order:false, stats:pseudo                 | 147 Bytes | N/A  |
+-------------------------------+---------+---------+-----------+------------------------+-------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+------------------------------------------------+-----------+------+
4 rows in set (0.00 sec)
```